language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/VariableCompletionItemBuilder.java

```diff
@@ -43,7 +44,9 @@ public static CompletionItem build(VariableSymbol varSymbol, String label, Strin
         CompletionItem item = new CompletionItem();
         item.setLabel(label);
         String[] delimiterSeparatedTokens = (label).split("\\.");
-        item.setInsertText(delimiterSeparatedTokens[delimiterSeparatedTokens.length - 1]);
+        String insertText = delimiterSeparatedTokens[delimiterSeparatedTokens.length - 1];
+        insertText = CommonUtil.escapeEscapeCharsInIdentifier(insertText);
+        item.setInsertText(insertText);
```

Comment:
What if we handle this in a more central position? For example, before returning the item list? Otherwise we will have to consider this fix in few places. For example, for the constants and functions.

Reply:
Agreed. Will do so.